Favorites screen, pepatadicozze.jpg

fourlastor opened the pull request:
Base impl for favorites screen. Right now I'm using schedule's screen views, this is mainly the service layer.

![device-2017-03-17-124714](https://cloud.githubusercontent.com/assets/1263058/24043746/06bd53bc-0b10-11e7-8dc5-f5777176f8bc.gif)

What's missing:

* Date headers
* Views if we decide we want to have different ones
* Theme (right now it's pink)

fourlastor committed:
Favorites screen, pepatadicozze.jpg

rock3r commented:
it'd be nice to have a screenshot :P

rock3r commented on app/src/main/java/net/squanchy/favorites/FavoritesPageView.java:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    private void onEventClicked(Event event) {
+        analytics.trackItemSelected(ContentType.SCHEDULE_ITEM, event.id());
```
Next PR: create a `FAVORITE_ITEM` type plz

rock3r commented on app/src/main/java/net/squanchy/favorites/FavoritesPageView.java:
```diff
@@ -0,0 +1,121 @@
+    }
+
+    private void handleProximityEvent(ProximityEvent proximityEvent) {
+        // TODO do something with the event, like showing feedback or opening an event detail
```
I think we should move this to the activity level instead

fourlastor replied:
Yep agreed

fourlastor replied:
252059f

rock3r commented on app/src/main/java/net/squanchy/favorites/FavoritesPageView.java:
```diff
@@ -0,0 +1,121 @@
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(favorites -> updateWith(favorites, this::onEventClicked)));
+
+        subscriptions.add(
```
This should be wrapped by the feature flag check

fourlastor replied:
252059f

rock3r commented on app/src/main/java/net/squanchy/favorites/domain/view/Favorites.java:
```diff
@@ -0,0 +1,16 @@
+
+@AutoValue
+public abstract class Favorites {
+    public static Favorites create(List<Event> events) {
```
cmd-alt-l plz

fourlastor replied:
e52f80d

rock3r commented on app/src/main/java/net/squanchy/schedule/view/EventsAdapter.java:
```diff
@@ -24,15 +24,12 @@
         int TYPE_OTHER = 1;
     }
 
-    private final Context context;
```
💯

fourlastor committed:
Format for the format :goat:

fourlastor committed:
Merge branch 'SQ-135/feature-favorites-screen' into SQ-135/base-favorites

fourlastor committed:
Stop radar in activity.onStop

fourlastor committed:
Add data/ manually to paths

Since we need to read also from /user automating it in observeChildAndEmit would break things

fourlastor committed:
Moved observing proximity events inside HomeActivity

fourlastor committed:
Restore flag after stopping the radar

rock3r reviewed (approved):
LGTM 👍
